### PR TITLE
Create erc20 token

### PR DESCRIPTION
**Pull request by EdgarO-Solicy**

### CODE REVIEW

File: erc_20.sol [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.7.0 <0.9.0;
+import "https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.0.0/contracts/token/ERC20/ERC20.sol";
+
+contract SolicyCoinERC20 is ERC20 {
+    mapping(address => uint256) balances;
```
Comment (Shitikyan) [inline erc_20.sol]: you have this in ERC20 parent class

File: erc_20.sol [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    mapping(address => bool) blackList;
+    uint8 _decimal;
+    uint8 _fee;
+    address public owner;
+    uint256 totalSupply_;
```
Comment (Shitikyan) [inline erc_20.sol]: you have this in ERC20 parent class

File: erc_20.sol [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    string public constant name = "Solicy";
+    string public constant symbol = "Sol";
+    uint8 public constant decimals = 18;
```
Comment (Shitikyan) [inline erc_20.sol]: These three arguments  should be passed via constructor

File: erc_20.sol [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    constructor(uint256 total, uint8 __decimal, uint8 __fee) ERC20("Solicy", "SOL"){
```
Comment (Shitikyan) [inline erc_20.sol]: You should pass your variables to ERC20 constructor, you are doing same thing twice

File: erc_20.sol [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+    function addToBlackList(address user) external {
+        blackList[user] = true;
+    }
+
+    function delFromBlackList(address user) external {
+        delete blackList[user];
+    }
+
+    function burn(address account, uint256 amount) external virtual {
```
Comment (Shitikyan) [inline erc_20.sol]: Here you should use modifier only token owner, in this case anyone can burn other person tokens

File: erc_20.sol [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        _burn(account, amount);
+    }
+
+    function mint() pure public {
+        require(false, "Mint action blocked in this contract");
+    }
+
+    function transfer(address recipient, uint256 amount) public virtual override returns (bool) {
+        require(!blackList[recipient], "some message");
```
Comment (Shitikyan) [inline erc_20.sol]: you need to implement transfer from with same logic for blacklist

File: erc_20.sol [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        _transfer(_msgSender(), recipient, amount);
+        return true;
+    }
+
+    function decimals() public view override returns (uint8) {
+        return _decimal;
+    }
+
+    function setFee(uint8 fee) public {
+        require(msg.sender == owner, "Ownable: You are not the owner, Bye.");
```
Comment (Shitikyan) [inline erc_20.sol]: you olready have modifier for this checking

File: erc_20.sol [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        _mint(msg.sender, 10 ** 10);
+    }
+
+    modifier onlyOwner {
+        require(msg.sender == owner);
+        _;
+    }
+
+    modifier tockenOwner {
+        require(balances[msg.sender] > 0);
```
Comment (Shitikyan) [inline erc_20.sol]: this modifier should take address and check if that address is msg.sender

File: erc_20.sol [hunk 9]
```diff
@@ -51,16 +42,12 @@ contract SolicyCoinERC20 is ERC20 {
 
     function transfer(address recipient, uint256 amount) public virtual override returns (bool) {
         require(!blackList[recipient], "some message");
-        _transfer(_msgSender(), recipient, amount);
+        transferFrom(_msgSender(), recipient, amount);
```
Comment (GagMirz) [inline erc_20.sol]: not here, you need to have also method for transferFrom

Comment (GagMirz) [reply erc_20.sol]: Not just call transferFrom in transfer

**Review by GagMirz — APPROVED** [review]

**Review by Shitikyan — APPROVED** [review]

**Pull request merged**